Sources/Common/Core/Math/index.js

```diff
@@ -2135,33 +2138,25 @@ export function float2CssRGBA(rgbArray) {
 }
 
 export function rotateVector(vectorToBeRotated, axis, angle) {
-  const v = [...vectorToBeRotated];
-  const l = [...axis];
-
-  normalize(v);
-  normalize(l);
-
-  const u = Math.sin(angle);
-  const w = 1.0 - Math.cos(angle);
-
-  const rotatedVector = [0, 0, 0];
-
-  rotatedVector[0] =
-    v[0] * (1 - w * (l[2] * l[2] + l[1] * l[1])) +
-    v[1] * (-u * l[2] + w * l[0] * l[1]) +
-    v[2] * (u * l[1] + w * l[0] * l[1]);
-
-  rotatedVector[1] =
-    v[0] * (u * l[2] + w * l[0] * l[1]) +
-    v[1] * (1 - w * (l[0] * l[0] + l[2] * l[2])) +
-    v[2] * (-u * l[0] + w * l[1] * l[2]);
-
-  rotatedVector[2] =
-    v[0] * (-u * l[1] + w * l[0] * l[2]) +
-    v[1] * (u * l[0] + w * l[1] * l[2]) +
-    v[2] * (1 - w * (l[1] * l[1] + l[0] * l[0]));
+  const matrix = vtkMatrixBuilder
+    .buildFromRadian()
+    .identity()
+    .rotate(angle, axis)
+    .getMatrix();
+
+  const newVector = [
+    vectorToBeRotated[0] * matrix[0] +
+      vectorToBeRotated[1] * matrix[4] +
+      vectorToBeRotated[2] * matrix[8],
+    vectorToBeRotated[0] * matrix[1] +
+      vectorToBeRotated[1] * matrix[5] +
+      vectorToBeRotated[2] * matrix[9],
+    vectorToBeRotated[0] * matrix[2] +
+      vectorToBeRotated[1] * matrix[6] +
+      vectorToBeRotated[2] * matrix[10],
```

Comment:
because you do only rotations, can't you simply do `vtkMatrixBuilder....apply(vectorToBeRotated)` ?